sklearn/semi_supervised/label_propagation.py

```diff
@@ -264,13 +280,20 @@ def fit(self, X, y):
             l_previous = self.label_distributions_
             self.label_distributions_ = safe_sparse_dot(
                 graph_matrix, self.label_distributions_)
+
+            if self.variant == 'propagation':
+                normalizer = np.sum(
+                    self.label_distributions_, axis=1)[:, np.newaxis]
+                self.label_distributions_ /= normalizer
+
             # clamp
             self.label_distributions_ = np.multiply(
                 clamp_weights, self.label_distributions_) + y_static
```

Comment:
Hmm. In the label prop case, we could just be using `np.where(unlabelled, self.label_distributions, y_static)` unless I'm mistaken